docs/src/main/asciidoc/getting-started-dev-services.adoc

```diff
@@ -0,0 +1,326 @@
+do you know anything is being written to the database?
+
+Add a second REST method to `GreetingResource`.
+
+----
+@GET
+@Path("names")
+@Produces(MediaType.TEXT_PLAIN)
+public String names() {
+List<Greeting> greetings = Greeting.listAll();
+   String names = greetings.stream().map(g-> g.name)
+   .collect(Collectors.joining (", "));
+   return "I've said hello to " + names;
+}
+----
+
+To try it out, visit http://localhost:8080/hello?name=Bloom, and then http://localhost/hello/names.
+
+You should see the following message: "I've said hello to Bloom".
+
+== Dev services
+
+Reading and writing to the database seems to be working well, but that's a bit unexpected.
+Where did a PostgresQL database come from? You didn't set anything up.
+
+The database is being managed using xref:{docfile}/dev-services.adoc[dev services].
+Dev services take care of stopping and starting services needed by your application.
+Because you
+included the `postgresql-jdbc` dependency, the database is a containerised PostGresQL database.
+If you'd added `postgresql-mysql` insead, you would have gotten a containerised MySQL database.
+
+If you like, use your container tool to see what containers are running.
+For example, if you're using Docker, run `docker ps`, and for podman, run `podman ps`.
+You'll see something like the following:
+
+----
+ff88dcedd899  docker.io/library/postgres:14  postgres -c fsync...  1 hour ago  Up 1 hour  0.0.0.0:34789->5432/tcp  nostalgic_bassi
+----
+
+Stop Quarkus and run `docker ps` again.
+You should see nothing running.
+Quarkus will automatically stop the container when your application stops.
+
+=== Initialising services
+
+If you play with your code some more, you may notice that sometimes, after making an application change, http://localhost/hello/names doesn't list any names.
+What's going on? By default, in dev mode, with a dev services database,
+ Quarkus configures Hibernate database generation to be `drop-and-create`.
+ See the xref:{docfile}/hibernate-orm.adoc#quarkus-hibernate-orm_quarkus.hibernate-orm.database-database-related-configuration[Hibernate configuration reference] for more details.
+ If a code change triggers an application restart, the database tables
+ will be dropped (deleted) and then re-created.
+
+This is convenient, but what if you'd prefer the database to always have content?
+That would make testing easier.
+If you provide an `import.sql` file, Quarkus will use that to initialise
+the database on each start.
+
+1. Make a `src/main/resources/import.sql` file in your project
+2. Add the following SQL statements:
+
+----
+INSERT INTO Greeting(id, name)
+VALUES (nextval('Greeting_SEQ'), 'Alice');
+INSERT INTO Greeting(id, name)
+VALUES (nextval('Greeting_SEQ'), 'Bob');
+----
+
+Now, try stopping and starting your application, and then visiting http://localhost:8080/hello/names.
+You'll see that Alice and Bob are always included in the list of names.
+
+== Controlling dev services
```

Comment:
I'm wondering if the rest of the document should be in this guide or a more "reference" guide.

Reply:
To me, it feels like we're leaving people with an incomplete picture if we don't include it; we've told them about dev services, but not giving them any hints about what happens if they don't want to use dev services. They might think they can't use a real database with Quarkus, and I'm not sure we want them to have to go read the both configuration reference and the dev services reference to figure out that it's possible. "Use profiles to conditionally enable dev services" would span the two references, which is an extra barrier to people figuring out the idea on their own.